File: 프로그래머스/lv1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.cpp
using namespace std;

long long solution(int price, int money, int count)
{
    long long answer = 0;

    for (int i = 1; i <= count; i++)
        answer += price * i;
    if (money - answer < 0)
        return answer - money;
    else
        return 0;
    
    return -1;
}

File: 프로그래머스/1/82612. 부족한 금액 계산하기/부족한 금액 계산하기.java
class Solution {
    public long solution(int price, int money, int count) {
        long sum = 0;
        for (int i = 0, init = price; i < count; i++) {
            sum += price;
            price += init;
        }
        return sum < money ? 0 : sum - money;
    }
}